### PR TITLE
Adds equality method for TDS item and tests

### PR DESCRIPTION
**Pull request by MDSilber**

### CODE REVIEW

File: ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceItem.mm [hunk 1]
```diff
@@ -12,6 +12,8 @@
 #import "CKTransactionalComponentDataSourceItemInternal.h"
 
 #import "CKComponentLayout.h"
+#import "CKEqualityHashHelpers.h"
```
Comment (adamjernst) [inline ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceItem.mm]: unused?

File: ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceItem.mm [hunk 2]
```diff
@@ -12,6 +12,8 @@
...
+#import "CKMacros.h"
```
Comment (adamjernst) [inline ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceItem.mm]: what here is used?

File: ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceItem.mm [hunk 3]
```diff
@@ -37,4 +39,18 @@ - (instancetype)initWithLayout:(const CKComponentLayout &)layout
   return _layout;
 }
 
+- (BOOL)isEqual:(id)object
+{
+  if (![object isKindOfClass:[CKTransactionalComponentDataSourceItem class]]) {
+    return NO;
+  } else {
+    return [_model isEqual:((CKTransactionalComponentDataSourceItem *)object).model];
```
Comment (adamjernst) [inline ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceItem.mm]: making this based on model equality is a bit odd to me.

**Comment by adamjernst** [comment]
Is there a good reason why we couldn't simply create a helper that does this externally (e.g. using a `NSComparator`) and leave the core untouched?

**Comment by MDSilber** [comment]
@adamjernst this is used by OCMock to compare object equality, so unfortunately we need to override isEqual:

**Comment by adamjernst** [comment]
@MDSilber  so don't use AssertEqualObjects. Use your own comparator and assert the result is YES.

**Pull request closed without merging**